test/String.js

```diff
@@ -40,6 +40,18 @@ describe('String', function() {
       const string = new StringT(null, 'utf8');
       assert.equal(string.fromBuffer(Buffer.from('🍻')), '🍻');
     });
+
+    it('should decode two-byte null-terminated string for utf16le', function() {
+      const stream = new DecodeStream(Buffer.from('🍻\x00', 'utf16le'));
+      const string = new StringT(null, 'utf16le');
+      assert.equal(string.decode(stream), '🍻');
+      assert.equal(stream.pos, 6);
+    });
+
+    it('should decode remainder of buffer when null-byte missing, utf16le', function() {
+      const string = new StringT(null, 'utf16le');
+      assert.equal(string.fromBuffer(Buffer.from('🍻', 'utf16le')), '🍻');
```

Comment:
Are strings in Javascript not always null-terminated?

Reply:
Strings in Javascript may be null terminated in the internal implementation, but this would never be visible to script itself. Strings have a `.length` property and attempts to access past the end of the string, e.g. `'123'[3]`, return `undefined`.

Reply:
It's also worth noting here that `StringT` is accessing binary data so it's providing a Javascript view of the binary data as a string. You can see an example elsewhere in the file where null termination is visible with `\x00` in the assertion.